enterprise/ha/src/main/java/org/neo4j/kernel/impl/ha/ClusterManager.java

```diff
@@ -992,6 +992,13 @@ public RepairKit shutdown( HighlyAvailableGraphDatabase db )
             members.remove( serverId );
             life.remove( db );
             db.shutdown();
+            // Sleep a little to help ensure that the shutdown thread has completed before we return
+            try
+            {
+                Thread.sleep( 50L );
```

Comment:
I wonder... a call to `db.shutdown()` is there a possibility that the db isn't actuallt shut down completely when it returns? Are we hiding an issue by sleeping here? Condition-less sleeping in a test to fix a problem doesn't fix a problem, it makes it less likely to happen.